src/test/scala/com/fulcrumgenomics/fastq/DemuxFastqsTest.scala

```diff
@@ -320,6 +320,97 @@ class DemuxFastqsTest extends UnitSpec with OptionValues with ErrorLogLevel {
+    DemuxResult(sampleInfo = sampleInfos(0), numMismatches = 0, records = Seq(demuxRecord))
+  }
+
+  "DemuxResult.countMismatches" should "mask bases that are less than or equal to the quality threshold" in {
+    val detector = new QualityEncodingDetector()
+    val qualities = "?????" + Range.inclusive(2, 40).map(q => (q + 33).toChar).mkString
+    val bases = Seq("GGGGG", "A"*39)
+    val expectedBases = "GGGGG" + "N"*9 + "A"*30
+
+    qualities.foreach(q => detector.add(q))
+    detector.compatibleEncodings(0).toString shouldBe "Standard"
+
+    val demuxResult = makeDemuxRecord(bases = bases.mkString, quals = qualities)
+    val output = demuxResult.maskLowQualityBases(threshold = 10, qualityEncoding = detector.compatibleEncodings(0)).records(0).bases
+
+    output.length shouldEqual qualities.length
+    output shouldEqual expectedBases.mkString
+  }
+
+  it should "mask no bases if the quality threshold is 0" in {
+    val detector = new QualityEncodingDetector()
+    val qualities = "?????" + Range.inclusive(2, 40).map(q => (q + 33).toChar).mkString
+    val bases = Seq("GGGGG", "A"*39)
+
+    qualities.foreach(q => detector.add(q))
+    detector.compatibleEncodings(0).toString shouldBe "Standard"
+
+    val demuxResult = makeDemuxRecord(bases = bases.mkString, quals = qualities)
+    val output = demuxResult.maskLowQualityBases(threshold = 0, qualityEncoding = detector.compatibleEncodings(0)).records(0).bases
+
+    output.length shouldEqual qualities.length
+    output shouldEqual bases.mkString
+  }
+
+  def testEndToEndWithQualityThreshold(threshold: Int = 0, threads: Int = 1): Seq[FastqRecord] = {
+    // Build the FASTQ
+    val output: DirPath = outputDir()
+
+    val metrics = makeTempFile("metrics", ".txt")
+    val structures = Seq(ReadStructure("17B139T"))
+
+    val metadata = sampleSheetPath
+
+    new DemuxFastqs(inputs=Seq(fastqPathSingle), output=output, metadata=metadata,
+      readStructures=structures, metrics=Some(metrics), maxMismatches=2, minMismatchDelta=3,
+      threads=threads, outputType=Some(OutputType.Fastq), qualityThreshold = threshold).execute()
+
+    val sampleInfo = toSampleInfos(structures).head
+    val sample = sampleInfo.sample
+
+    val prefix = toSampleOutputPrefix(sample, sampleInfo.isUnmatched, false, output, UnmatchedSampleId)
+
+    val fastq = PathUtil.pathTo(s"${prefix}.fastq.gz")
+    FastqSource(fastq).toSeq
+}
+
+  "DemuxFastqs" should "run end to end and return the same bases when the threshold is 0 for 1 thread" in {
+    val records = testEndToEndWithQualityThreshold(threshold = 0, threads = 1)
+    records.length shouldBe 1
+    records(0).bases.length shouldBe 39
+    records(0).bases shouldEqual "A"*39
+  }
+
+  "DemuxFastqs" should "run end to end and replace any bases below a specified quality threshold for 1 thread" in {
```

Comment:
For the first test you put the label (e.g. "DemuxFastqs") but for subsequent ones you want to use just `it`